Library/Homebrew/download_strategy.rb

```diff
@@ -340,29 +342,29 @@ def parse_basename(url, search_query: true)
         end
       end
 
-      if uri.query && search_query
-        "#{uri.path}?#{uri.query}"
-      else
-        uri.path
+      if (uri_path = uri.path)
+        components[:path] = uri_path.split("/").map do |part|
+          URI::DEFAULT_PARSER.unescape(part).presence
+        end.compact
+      end
+
+      if search_query && (uri_query = uri.query)
```

Comment:
```suggestion
      if search_query.present? && (uri_query = uri.query.presence)
```

Reply:
`search_query` is a boolean so I'll address that part by adding a type signature instead.